### PR TITLE
Feature: Create the controller and routes for League

### PR DESCRIPTION
**Pull request by JesusGo115**
## Summary by CodeRabbit

- **New Features**
	- Introduced a new set of routes for managing leagues within groups.
	- Added a configuration block for MySQL testing settings.

- **Enhancements**
	- Improved logging and response handling in middleware for better debugging and user feedback.

- **Tests**
	- Implemented comprehensive test cases for the new League Routes functionality.

### CODE REVIEW

File: server/routes/league.routes.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+"use strict";
+
+const express = require("express");
+const router = express.Router();
+const LeagueController = require("../controllers/league.controller");
+
+router.post("/create", LeagueController.createLeague);
+router.get("/getByGroupId", LeagueController.getLeagueByGroupId);
+router.post("/update", LeagueController.updateLeague);
```
Comment (saulRM10) [inline server/routes/league.routes.js:9]: I think for updates we should go with `PATCH` , i need to make that change on my pr as well

File: server/controllers/league.controller.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+      await League.build(newLeague).validate();
+      const result = await League.create(newLeague);
+
+      return res.status(201).json({
+        success: true,
+        data: result,
+      });
+    } catch (error) {
+      // const validationErrors = error.errors?.map((err) => ({
```
Comment (saulRM10) [inline server/controllers/league.controller.js:55]: remove the comments

File: server/routes/league.routes.js (hunk 1 above)
Comment (JesusGo115) [reply server/routes/league.routes.js]: Made the change to the route and made it PATCH instead.

File: server/controllers/league.controller.js (hunk 2 above)
Comment (JesusGo115) [reply server/controllers/league.controller.js]: Comments have been removed.

File: server/__tests__/routeTests/league.routes.test.js [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+  });
+
+  it("should return an error if league not found PATCH /patch", async () => {
+    const nonExistentLeagueId = 9999; 
+
+    const response = await request(app)
+      .patch(`/league/${nonExistentLeagueId}`)
+      .send({ name: "Joe Mo Mah Inc." });
+
+    expect(response.status).toBe(500);
```
Comment (saulRM10) [inline server/__tests__/routeTests/league.routes.test.js:120]: what is the error message ? Can you add one

File: server/controllers/league.controller.js [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+          id: req.params['id'],
+        },
+      });
+
+      if (deletedLeague === 0) {
+        return res
+          .status(500)
+          .json({ error: "No league found with the given ID" });
+      }
+      return res.status(200).json({ success: true, message: "Delete league successfully" });
```
Comment (saulRM10) [inline server/controllers/league.controller.js:101]: for deleted it should be '204'

File: server/controllers/league.controller.js [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+        where: {
+          id: req.params['id'],
+        },
+      });
+
+      if (!currentLeague) {
+        throw { field: "name", message: "League with given ID was not found" };
+      }
+
+      currentLeague.group_id = req.body.group_id || currentLeague.group_id;
```
Comment (saulRM10) [inline server/controllers/league.controller.js:73]: user should not be able to update the group id

**Review by saulRM10 — APPROVED** [review]
looks good

File: server/controllers/league.controller.js [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+      currentLeague.description = req.body.description || currentLeague.description;
+
+      await isNameUniqueWithinGroup(currentLeague.group_id, currentLeague.name);
+
+      await currentLeague.save();
+
+      return res.status(200).json({ success: true, data: currentLeague });
+    } catch (error) {
+      next(error);
+      console.log("in error stuffs for update");
```
Comment (saulRM10) [inline server/controllers/league.controller.js:84]: remove console.log used for debugging

File: server/controllers/league.controller.js [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+
+  var deleteLeague = async function (req, res, next) {
+    try {
+      let deletedLeague = await League.destroy({
+        where: {
+          id: req.params['id'],
+        },
+      });
+
+      console.log(deletedLeague);
```
Comment (saulRM10) [inline server/controllers/league.controller.js:96]: delete console.log here

File: server/controllers/league.controller.js [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+
+      if (deletedLeague === 0) {
+        throw new Error("No league found with the given ID");
+       }
+
+      // TODO: for deleted it should be '204' (Donzo)
```
Comment (saulRM10) [inline server/controllers/league.controller.js]: remove console logs

Comment (JesusGo115) [reply server/controllers/league.controller.js]: I removed the console.logs within the code.

File: server/controllers/league.controller.js (hunk 7 above)
Comment (JesusGo115) [reply server/controllers/league.controller.js]: Deleted

File: server/controllers/league.controller.js (hunk 6 above)
Comment (JesusGo115) [reply server/controllers/league.controller.js]: Deleted

**Review by saulRM10 — APPROVED** [review]
thanks for making all the changes great job, looks good

File: server/middlewares.js [hunk 9]
```diff
@@ -23,12 +23,16 @@ const Middlewares = {
     const statusCode = res.statusCode !== 200 ? res.statusCode : 500;
     res.status(statusCode);
 
+    console.log("In middleware");
+
+    console.log(err.message);
```
[the remark this answers is not in the document]
Comment (JesusGo115) [reply server/middlewares.js]: Deleted the console logs.

File: server/controllers/league.controller.js (hunk 5 above)
Comment (JesusGo115) [reply server/controllers/league.controller.js]: Deleted that functionality.

File: server/controllers/league.controller.js [hunk 10]
```diff
@@ -0,0 +1,158 @@
...
+
+      if (deletedLeague === 0) {
+        return res
+          .status(404)
+          .json({ error: "No league found with the given ID" });
+      }
+      return res.status(200).json();
+    } catch (error) {
+      return res.status(500).json({ error: "Failed to delete league" });
+    }
```
[the remark this answers is not in the document]
Comment (JesusGo115) [reply server/controllers/league.controller.js]: Deleted the duplicates.

File: server/controllers/league.controller.js [hunk 11]
```diff
@@ -0,0 +1,158 @@
...
+      currentLeague.description =
+        req.body.description || currentLeague.description;
+
+      await currentLeague.save();
+
+      return res.status(200).json({ success: true, data: currentLeague });
+    } catch (error) {
+      return res.status(500).json({ error: "Failed to update League" });
+    }
+  };
```
[the remark this answers is not in the document]
Comment (JesusGo115) [reply server/controllers/league.controller.js]: Deleted the duplicate.

File: server/controllers/league.controller.js [hunk 12]
```diff
@@ -0,0 +1,158 @@
...
+      const result = await League.create(newLeague);
+
+      return res.status(201).json({
+        success: true,
+        data: result,
+      });
+    } catch (error) {
+      next(error);
+    }
+  };
```
[the remark this answers is not in the document]
Comment (JesusGo115) [reply server/controllers/league.controller.js]: Updated this function.

File: server/controllers/league.controller.js (hunk 4 above)
Comment (JesusGo115) [reply server/controllers/league.controller.js]: Changed it to 204.

File: server/controllers/league.controller.js [hunk 13]
```diff
@@ -0,0 +1,158 @@
...
+
+      if (deletedLeague === 0) {
+        return res
+          .status(500)
+          .json({ error: "No league found with the given ID" });
+      }
+      return res.status(200).json({ success: true, message: "Delete league successfully" });
+    } catch (error) {
+      next(error)
+    }
```
[the remark this answers is not in the document]
Comment (JesusGo115) [reply server/controllers/league.controller.js]: Changed it to 204.

**Pull request merged**